lib/logstash/inputs/tcp.rb

```diff
@@ -164,11 +171,14 @@ def handle_socket(socket, client_address, client_port, output_queue, codec)
       end
     end
   rescue EOFError
+    @metric_errors.increment(:connection)
```

Comment:
Agree, `EOF` should be tracked as a connection error, same thing with SSLError. 
Theses kinds of errors are so common in a **normal** lifecycle of an application and tracking them will give probably result in new issues created in the tracker.

I remember one time, when I decided to log theses error in the beat plugin with the `error` level, people using LB in front of Logstash were creating issues asking *what is going on*, only to discover that the keep alive check were causing them ;)